### PR TITLE
Replace JSON configs with YAML

### PR DESCRIPTION
**Pull request by mlevesquedion**
Now that YAML is the recommended way to configure the analyzer, in this PR I propose removing (almost) all traces of JSON from our codebase.

I think having every configuration file be YAML has two main benefits:
1. Consistency
2. YAML is easier to read and edit (this may be personal preference talking)

The only remaining traces of JSON are in `config/regexp/*.go`. This code uses `json` Unmarshaling, which is compatible with the `yaml` package. I think it's fine to leave it like that.

- [x] Tests pass
- (N/A) [ ] Running against a large codebase such as [Kubernetes](https://github.com/kubernetes/kubernetes) does not error out.
- (N/A) [ ] Appropriate changes to README are included in PR

### CODE REVIEW

File: configuration/README.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ As just two examples, this may be used to avoid analyzing test code, or to suppr
 
 The following configuration could be used to identify possible instances of credential logging in Kubernetes.
 
-[example-config.json](example-config.json)
+[example-config.yaml](example-config.yaml)
```
Comment (mlevesquedion) [inline configuration/README.md:76]: The actual renaming was done in #163. This is just modifying the pointer to point to the new name.

File: configuration/example.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@ GB=$(go env GOBIN) GB=${GB:-$(go env GOPATH)/bin}
 # go vet dislikes relative pathing.
 SCRIPT_DIR=$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )
 # analysis arguments
-CONFIG_FILE=${SCRIPT_DIR}/example-config.json
+CONFIG_FILE=${SCRIPT_DIR}/example-config.yaml
```
Comment (mlevesquedion) [inline configuration/example.sh:8]: The actual renaming was done in #163. This is just modifying the pointer to point to the new name.

File: .travis.yml [hunk 3]
```diff
@@ -28,12 +28,11 @@ script:
   )
 - >
   echo "Checking for license headers..." && (
-    FILES=$(grep -rL 'https://www.apache.org/licenses/LICENSE-2.0' cmd internal pkg | grep -vE ".(json|ssa)")
+    FILES=$(grep -rL 'https://www.apache.org/licenses/LICENSE-2.0' cmd internal pkg | grep -vE ".(yaml|ssa)")
```
Comment (PurelyApplied) [inline .travis.yml:31]: Presumably just swept up in a find-and-replace, but the advantage of comments in YAML is that we can and should include the license header.

**Review by PurelyApplied — CHANGES_REQUESTED** [review]
Looks good to me.  Thank you and your apparently enormous well of patience for wading through these.

Minor issue in the Travis, but LGTM overall.

**Comment by mlevesquedion** [comment]
I must confess that I am actually very impatient. Mostly I just ran this command:

```bash
for f in $(fd json); do
    vim $f
done
```

And I had https://www.json2yaml.com/ open on the side.

File: .travis.yml (hunk 3 above)
Comment (mlevesquedion) [reply .travis.yml]: SG. Done.

**Review by PurelyApplied — APPROVED** [review]

**Pull request merged**